utils/response-rejection/src/index.ts

```diff
@@ -1,63 +1,156 @@
 import api, { Flatfile } from '@flatfile/api'
+import { processRecords } from '@flatfile/util-common'
 
-export interface ResponseRejection {
+export interface RejectionResponse {
   id: string
-  sheets: ResponseRejectionSheet[]
+  message?: string
+  deleteSubmitted?: boolean
+  sheets: SheetRejections[]
 }
 
-export interface ResponseRejectionSheet {
-  id: string
-  name?: string
-  rejectedRecords: ResponseRejectionRecord[]
+export interface SheetRejections {
+  sheetId: string
+  rejectedRecords: RecordRejections[]
 }
 
-export interface ResponseRejectionRecord {
+export interface RecordRejections {
   id: string
   values: { field: string; message: string }[]
 }
 
 export async function responseRejectionHandler(
-  responseRejection: ResponseRejection
-): Promise<number> {
+  responseRejection: RejectionResponse
+): Promise<void | Flatfile.JobCompleteDetails> {
   let totalRejectedRecords = 0
 
-  const results = await Promise.all(
-    responseRejection.sheets.map((sheet) => updateSheet(sheet))
-  )
-  totalRejectedRecords = results.reduce((acc, val) => acc + val, 0)
+  for (const sheet of responseRejection.sheets || []) {
+    const count = await updateSheet(sheet, responseRejection.deleteSubmitted)
+    totalRejectedRecords += count
+  }
+
+  const message = responseRejection.message ?? getMessage(totalRejectedRecords)
+  const next = getNext(totalRejectedRecords)
+
+  return {
+    outcome: {
+      buttonText: 'Close',
+      heading: totalRejectedRecords > 0 ? 'Rejected Records' : 'Success!',
+      acknowledge: true,
+      ...(next && !responseRejection.deleteSubmitted && { next }),
+      message,
+    },
+  }
+}
+
+function getMessage(totalRejectedRecords) {
+  return totalRejectedRecords > 0
+    ? `During the data submission process, ${totalRejectedRecords} records were rejected. Please review and correct these records before resubmitting.`
+    : 'The data has been successfully submitted without any rejections. This task is now complete.'
+}
 
-  return totalRejectedRecords
+function getNext(totalRejectedRecords): Flatfile.JobOutcomeNext | undefined {
+  return totalRejectedRecords > 0
+    ? {
+        type: 'url',
+        url: '?searchField=submissionStatus&searchValue=rejected',
+        label: 'See rejections',
+      }
```

Comment:
Change this to use : `{ next: { type: 'id', id: 'us_sp_12345', query: 'filter=error'}}`